vhdl_ls/src/vhdl_server.rs

```diff
@@ -681,6 +682,36 @@ impl VHDLServer {
         })
     }
 
+    pub fn document_highlight(
+        &mut self,
+        params: &TextDocumentPositionParams,
+    ) -> Option<Vec<DocumentHighlight>> {
+        let source = self
+            .project
+            .get_source(&uri_to_file_name(&params.text_document.uri))?;
+
+        let ent = self
+            .project
+            .find_declaration(&source, from_lsp_pos(params.position))?;
+
+        Some(
+            self.project
```

Comment:
One small optional improvement here:
In your implementation, all references in all files are being searched. After that, only references concerning the current file are kept. This could be improved by just searching in one source file altogether.
For this purpose, the `DesignRoot` has a function `search_source`.
My suggested change is creating a new function (something like `find_all_references_in_source(...)` that only searches in the given source, so the last `filter_map` step could be replaced with a single `map` step.

Reply:
I thought something like that should have existed, but I did not find it, so I went for the simpler but less efficient method. I will have a look at your suggestion and try to implement it.